pkg/controller/webhook_test.go

```diff
@@ -263,3 +263,29 @@ func TestCanaryChecksum(t *testing.T) {
 	require.NotEqual(t, canary3sum, canary1sum)
 	require.NotEqual(t, canary4sum, canary1sum)
 }
+
+func TestCallWebhook_Retries(t *testing.T) {
+	retries := 1
+	failures := 0
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if failures <= retries-1 {
```

Comment:
```suggestion
		if failures < retries {
```

i feel like this is easier to read